src/coreclr/pal/src/misc/cgroup.cpp

```diff
@@ -52,25 +50,42 @@ class CGroup
     static char *s_memory_cgroup_path;
     static char *s_cpu_cgroup_path;
 
-    static const char *s_active_file_key_name;
-    static const char *s_inactive_file_key_name;
-    static const char *s_dirty_file_key_name;
-    static size_t s_active_file_key_length;
-    static size_t s_inactive_file_key_length;
-    static size_t s_dirty_file_key_length;
-
+    static const char **s_mem_stat_key_names;
+    static size_t *s_mem_stat_key_lengths;
+    static size_t s_mem_stat_n_keys;
+    
 public:
     static void Initialize()
     {
         s_cgroup_version = FindCGroupVersion();
         s_memory_cgroup_path = FindCGroupPath(s_cgroup_version == 1 ? &IsCGroup1MemorySubsystem : nullptr);
         s_cpu_cgroup_path = FindCGroupPath(s_cgroup_version == 1 ? &IsCGroup1CpuSubsystem : nullptr);
-        s_active_file_key_name = s_cgroup_version == 1 ? "total_active_file " : "active_file ";
-        s_inactive_file_key_name = s_cgroup_version == 1 ? "total_inactive_file " : "inactive_file ";
-        s_dirty_file_key_name = s_cgroup_version == 1 ? "total_dirty " : "file_dirty ";
-        s_active_file_key_length = strlen(s_active_file_key_name);
-        s_inactive_file_key_length = strlen(s_inactive_file_key_name);
-        s_dirty_file_key_length = strlen(s_dirty_file_key_name);
+        
+        if (s_cgroup_version == 1)
+        {
+            s_mem_stat_n_keys = 4;
+            s_mem_stat_key_names = new const char*[s_mem_stat_n_keys];
```

Comment:
Can you please make the s_mem_stat_key_names an array instead of allocating it via new? And the same for s_mem_stat_key_lengths.

Reply:
done